runtime/drivers/clickhouse/information_schema.go

```
@@ -113,6 +113,74 @@ func (i informationSchema) Lookup(ctx context.Context, db, schema, name string)
 	return tables[0], nil
 }
 
+func (i informationSchema) SizeOnDisk(ctx context.Context, tables []*drivers.Table) error {
+	if len(tables) == 0 {
+		return nil
+	}
+	conn, release, err := i.c.acquireMetaConn(ctx)
+	if err != nil {
+		return err
+	}
+	defer func() { _ = release() }()
+
+	var queryBuilder strings.Builder
+	queryBuilder.WriteString(`
+		SELECT 
+			database, 
+			table, 
+			SUM(bytes_on_disk) AS total_size_bytes
+		FROM system.parts
+		WHERE active = 1 AND (database, table) IN (
+	`)
+	args := make([]interface{}, 0, len(tables)*2)
+	placeholders := make([]string, 0, len(tables))
+
+	for _, table := range tables {
+		placeholders = append(placeholders, "(?, ?)")
+		args = append(args, table.DatabaseSchema, table.Name)
+	}
+
+	queryBuilder.WriteString(strings.Join(placeholders, ", "))
+	queryBuilder.WriteString(") GROUP BY database, table")
+
+	rows, err := conn.QueryxContext(ctx, queryBuilder.String(), args...)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
```

The caller may not have permission to access `system.parts`. To be consistent with the docstring, shouldn't it check for a permission error here and set `bytes_on_disk` to `-1` in that case?

---

I expected the caller to take a call on whether to handle error or ignore. I imagined that we could ignore the error in OLAP APIs for example but that may not be correct to do in API exposed as part of `InformationSchema`. 
It returns -1 already if it is a table.

---

Since we may have several places that try to get table sizes (across UI, CLI, reconcilers) and checking the error message with string matching is cumbersome, I think returning `-1` for "unknown" is a reasonable thing to do (and it's already described in the docstring). It also propagates nicely, e.g. even if no handling for `-1` is done, it will often be self-explanatory for the end user to see that number.